src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -76,15 +76,17 @@ func (c *controller) WaitAndUpdateNodesStatus() {
 		models.HostStatusError, models.HostStatusInstalled}
 	for {
 		time.Sleep(GeneralWaitInterval)
-		assistedInstallerNodesMap, err := c.ic.GetHosts(ignoreStatuses)
+		ctx := utils.GenerateRequestContext()
+		log := utils.RequestIDLogger(ctx, c.log)
+		assistedInstallerNodesMap, err := c.ic.GetHosts(ctx, log, ignoreStatuses)
```

Comment:
What?
Is this actually required (generating a request ID and setting a logger)?
Can't we get the request ID (generated by the service) from the response?

Reply:
yes, like the agent, we need to generate ID here and send it to assisted-service. 
if we won't send an id assisted-service will not generate one.